internal/controller/postgrescluster/pgbackrest_test.go

```diff
@@ -2317,6 +2317,198 @@ func TestReconcilePostgresClusterDataSource(t *testing.T) {
 	}
 }
 
+func TestGenerateRestoreJobIntent(t *testing.T) {
+	ctx := context.Background()
+
+	t.Run("empty", func(t *testing.T) {
+		generateRestoreJobIntent(ctx, &v1beta1.PostgresCluster{}, "", "",
+			[]string{}, []corev1.VolumeMount{}, []corev1.Volume{},
+			&v1beta1.PostgresClusterDataSource{}, &batchv1.Job{})
+	})
+
+	configHash := "hash"
+	instanceName := "name"
+	cmd := []string{"cmd", "blah"}
+	volumeMounts := []corev1.VolumeMount{{
+		Name: "mount",
+	}}
+	volumes := []corev1.Volume{{
+		Name: "volume",
+	}}
+	dataSource := &v1beta1.PostgresClusterDataSource{
+		// ClusterName/Namespace, Repo, and Options are tested in
+		// TestReconcilePostgresClusterDataSource
+		Resources: corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceStorage: resource.MustParse("1Gi"),
+			},
+		},
+		Affinity: &corev1.Affinity{
+			NodeAffinity: &corev1.NodeAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+					NodeSelectorTerms: []corev1.NodeSelectorTerm{{
+						MatchExpressions: []corev1.NodeSelectorRequirement{{
```

Comment:
🌱 YAML is a quick way to assign deep fields without having to get pointers and types correct.

```go
assert.NilError(t, yaml.Unmarshal([]byte(`{
	affinity: { nodeAffinity: { requiredDuringSchedulingIgnoredDuringExecution: {
		nodeSelectorTerms: [{ matchExpressions: [{ key: key, operator: Exists }] }],
	} } },
	resources: { requests: { storage: 1Gi } },
	tolerations: [{ key: key, operator: Exists }],
}`), &dataSource)

Reply:
If there are certain patterns emerging that we like more than others, lets keep in mind to discuss with the team as we plan our next sprint.